### PR TITLE
Enable read after update

### PR DESCRIPTION
**Pull request by andyfengHKU**

### CODE REVIEW

File: test/test_files/tinysnb/update_rel/set_read.test
Comment (ray6080) [inline test/test_files/tinysnb/update_rel/set_read.test:1]: Remove this test file or skip them.

File: test/test_files/tinysnb/exception/exception.test [hunk 1]
```diff
@@ -18,11 +18,6 @@ Runtime exception: Modulo by zero.
 ---- error
 Connection Exception: Query is empty.
 
--CASE ReadAfterUpdate2
```
Comment (ray6080) [inline test/test_files/tinysnb/exception/exception.test:21]: Revert this.

File: test/test_files/exceptions/binder/binder_error.test [hunk 2]
```diff
@@ -219,16 +219,6 @@ Binder exception: Expression hh has data type STRING but expect INT64. Implicit
 ---- error
 Binder exception: Expression 12 has data type INT64 but expect DATE. Implicit cast is not supported.
 
--LOG ReadAfterUpdate1
```
Comment (ray6080) [inline test/test_files/exceptions/binder/binder_error.test:222]: Revert here.

File: src/processor/operator/update/set_executor.cpp [hunk 3]
```diff
@@ -1,14 +1,31 @@
 #include "processor/operator/update/set_executor.h"
 
+using namespace kuzu::common;
+
 namespace kuzu {
 namespace processor {
 
 void NodeSetExecutor::init(ResultSet* resultSet, ExecutionContext* context) {
     nodeIDVector = resultSet->getValueVector(nodeIDPos).get();
```
Comment (ray6080) [inline src/processor/operator/update/set_executor.cpp:9]: Can we initialize `propertyVector` to nullptr in NodeSetExecutor's constructor?
Here we only need the `if` statement to set it to a valid value vector.

File: src/processor/operator/update/set_executor.cpp [hunk 4]
```diff
@@ -36,20 +59,40 @@ void MultiLabelNodeSetExecutor::set() {
...
+            writeToPropertyVector(propertyVector, lhsPos, rhsVector, rhsPos);
+        }
     }
 }
 
 void RelSetExecutor::init(ResultSet* resultSet, ExecutionContext* context) {
     srcNodeIDVector = resultSet->getValueVector(srcNodeIDPos).get();
     dstNodeIDVector = resultSet->getValueVector(dstNodeIDPos).get();
     relIDVector = resultSet->getValueVector(relIDPos).get();
+    if (propertyPos.dataChunkPos != INVALID_DATA_CHUNK_POS) {
```
Comment (ray6080) [inline src/processor/operator/update/set_executor.cpp:72]: Ditto here. We can init propertyVector to nullptr in RelSetExecutor's constructor.

**Review by ray6080 — APPROVED** [review]

File: src/include/processor/operator/update/set_executor.h [hunk 5]
```diff
@@ -24,19 +24,22 @@ class NodeSetExecutor {
 
 protected:
     DataPos nodeIDPos;
+    DataPos propertyPos;
     std::unique_ptr<evaluator::BaseExpressionEvaluator> evaluator;
 
     common::ValueVector* nodeIDVector;
+    common::ValueVector* propertyVector;
```
[the remark this answers is not in the document]
Comment (andyfengHKU) [reply src/include/processor/operator/update/set_executor.h:31]: Decide to use LHS and RHS with a comment in the code.

**Pull request merged**